website/content/posts/evaluating-custom-predicates.md

```diff
@@ -0,0 +1,196 @@
+    error := "Spam attestation does not set the 'spam' attribute"
+}
+
+_errors contains error if {
+    some attestation in _spam_attestations
+    got := attestation.statement.predicate.spam
+    got != _expected_spam
+    error := sprintf("The spam value %q is not the expected %q", [got, _expected_spam])
+}
+
+_spam_attestations := [attestation |
+    some attestation in input.attestations
+    attestation.statement.predicateType == "https://spam/v42"
+]
+```
+
+Next, we create a policy configuration, `policy.yaml`, to use the policy rules above.
+
+```yaml
+---
+sources:
+  - policy:
+      - /tmp/policies
+```
+
+Note: Change `/tmp/policies` to the absolute path to the directory where `spam.rego` was saved.
+
+Finally, let's validate our image conforms to the policy config.
+
+```bash
+ec validate image --public-key cosign.pub --policy policy.yaml --image $REPOSITORY:latest \
+  --output yaml --show-successes
+```
+
+```yaml
+components:
+- attestations:
+  - predicateBuildType: https://localhost/dummy-type
+    predicateType: https://slsa.dev/provenance/v0.2
+    signatures:
+    - keyid: ""
+      sig: MEUCIFMtiIiz0h9+zJpc5MfwavZ2/BIxuhIig5uoePcQ+nOHAiEAzSCKAOH5irMG1bG5HNkVzZLOyDOV3SiIIrU6YCTz668=
+    type: https://in-toto.io/Statement/v0.1
+  - predicateType: https://spam/v42
+    signatures:
+    - keyid: ""
+      sig: MEQCIHzmTK9YRU/PPfFjxRP6oSFNXyMIbAXEnQNP7GcCIjsbAiBVI8NtWYcvjg7/GmFC9Ce1e0XSh/mS5i5USHAX5I12tA==
+    type: https://in-toto.io/Statement/v0.1
+  containerImage: localhost:5000/ec-zero-to-hero@sha256:b5430d3d447434c795a508036e5046e41c009039be5b3f656f121c2426500d1e
+  name: Unnamed
+  signatures:
+  - keyid: ""
+    sig: MEQCIGIxpCboUYXF/fw6OuKmpM1Svi/0q+URD7oarLsji2+nAiBe3rgmWYOCa7sVpc2K5DKsef9hDigSlOHt6tl8v/8/JA==
+  source: {}
+  success: true
+  successes:
+  - metadata:
+      code: builtin.attestation.signature_check
+    msg: Pass
+  - metadata:
+      code: builtin.attestation.syntax_check
+    msg: Pass
+  - metadata:
+      code: builtin.image.signature_check
+    msg: Pass
+  - metadata:
+      code: spam.spam
+    msg: Pass
+ec-version: v0.3.2687-370a1a7
+effective-time: "2024-03-20T18:58:24.532182311Z"
+key: |
+  -----BEGIN PUBLIC KEY-----
+  MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEaxhfip26cuIfpDEVII0ilOHvFkee
+  igKbV5APr6OHYEstl9uJ8keON3VyzGfRB/2FUzr92J4ZN3YQUsTaiGc/HQ==
+  -----END PUBLIC KEY-----
+policy:
+  publicKey: cosign.pub
+  sources:
+  - policy:
+    - /tmp/admiring-mendel/policies
+success: true
+```
+
+Success!
+
+Change the value of the `_expected_spam` variable in the `spam.rego` file to any other value. Run
+the exact EC command again to see what a violation looks like.
+
+```yaml
+components:
+- ...
+  violations:
+  - metadata:
+      code: spam.spam
+    msg: The spam value "more spam" is not the expected "more eggs"
+...
+success: false
+```
+
+As we can see the Enterprise Contract can be used to perform advanced checks on any kind of
+attestation 👌. What will you use it for? 🤔
```

Comment:
Love it!

Reply:
Idea (maybe): Run it again with `--info` and see if the description shows up in the output.

Reply:
Also maybe one sentence to introduce the idea that EC does neat things with rego annotations with a link to https://enterprisecontract.dev/docs/ec-policies/authoring.html for more details.

Reply:
I added a fun sentence linking to the authoring docs. Good idea. 

I'm going to skip the suggestion about `--info` as I don't think it brings too much value to this particular topic.